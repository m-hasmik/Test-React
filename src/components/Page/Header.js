// @flow
import React from 'react'
import styled from 'styled-components'

import Title from '~/components/common/Title'
import { GRAY_MEDIUM, WHITE } from '~/constants/colors'

type Props = {
  title: string,
  actions: Array<any>
}

export const Header = ({ title, actions }: Props) => (
  <Wrapper>
    <Content data-testid="header-section">
      <Title>{title}</Title>
    </Content>
    <Actions data-testid="header-section">{actions}</Actions>
  </Wrapper>
)

const Wrapper = styled.div`
  align-items: flex-end;
  background: ${WHITE};
  border-bottom: 1px solid ${GRAY_MEDIUM};
  display: flex;
  padding: 1rem 2rem 1rem;
  width: 100%;
`

const Content = styled.div`
  flex: 1;
`

const Actions = styled.div`
  display: flex;
  justify-content: flex-end;
  flex: 1;
`
