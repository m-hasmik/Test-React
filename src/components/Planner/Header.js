// @flow
import React from 'react'
import moment from 'moment'
import styled from 'styled-components'
import Card from '~/components/common/Card'
import Paragraph from '~/components/common/Paragraph'
import { getCurrentWeek, getCurrentDays } from '~/helpers/dates'
import { DAYS_FILTER, WEEK } from '~/constants/dropdown'

const DAY_FORMAT: string = 'dddd'

type Props = {
  weekDay: moment,
  hasWeekendSchedule: boolean,
  getCurrentViewValue: number
}

const Header = ({
  weekDay,
  hasWeekendSchedule,
  getCurrentViewValue
}: Props) => {
  const currentDates =
    getCurrentViewValue === DAYS_FILTER[WEEK]
      ? getCurrentWeek(weekDay, { withWeekend: hasWeekendSchedule })
      : getCurrentDays(weekDay, getCurrentViewValue)
  const headers: Array<string> = currentDates.map(
    (date: Date): string => moment(date).format(DAY_FORMAT)
  )
  return (
    <Wrapper>
      {headers.map(header => (
        <Element key={header}>
          <Paragraph>{header}</Paragraph>
        </Element>
      ))}
    </Wrapper>
  )
}

const Wrapper = styled(Card)`
  display: flex;
  justify-content: space-evenly;
  margin: 0;
  width: 100%;
`

const Element = styled.div`
  flex: 1;
  text-align: center;
  min-width: 150px;

  ${Paragraph} {
    margin: 0;
  }
`

export default Header
