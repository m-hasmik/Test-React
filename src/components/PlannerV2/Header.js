// @flow
import React from 'react'
import styled from 'styled-components'
import Card from '~/components/common/Card'
import Paragraph from '~/components/common/Paragraph'

const Header = () => {
  const headers = [
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
    "Sunday"
  ]
  return (
    <Wrapper>
      {headers.map(header => (
        <Element key={header}>
          <Paragraph>{header}</Paragraph>
        </Element>
      ))}
    </Wrapper>
  )
}

const Wrapper = styled(Card)`
  display: flex;
  justify-content: space-evenly;
  margin: 0;
  width: 100%;
`

const Element = styled.div`
  flex: 1;
  text-align: center;
  min-width: 150px;

  ${Paragraph} {
    margin: 0;
  }
`

export default Header
