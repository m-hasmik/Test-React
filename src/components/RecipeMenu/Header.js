import styled from 'styled-components'

import Title from '~/components/common/Title'
import { GRAY_MEDIUM } from '~/constants/colors'

export const Header = styled.div`
  display: flex;
  align-items: center;
  border-bottom: 1px solid ${GRAY_MEDIUM};
  padding: 0.5rem;

  ${Title} {
    font-size: 0.9rem;
    margin-left: 0.5rem;
    overflow: hidden;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
  }
`
