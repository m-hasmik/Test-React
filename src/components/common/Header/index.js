import styled from 'styled-components'

import { BLACK } from '~/constants/colors'

const Header = styled.h2`
  font-family: 'Poppins', sans-serif;
  font-size: 1rem;
  font-weight: 200;
  color: ${BLACK};
  margin-bottom: 1rem;
`

export default Header
