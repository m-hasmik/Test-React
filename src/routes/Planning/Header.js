// @flow
import React from 'react'
import { withNamespaces } from 'react-i18next'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faFileExport } from '@fortawesome/free-solid-svg-icons'

import {
  INSTRUCTIONS_TEMPLATE,
  WEEKLY_MENU_TEMPLATE
} from '~/routes/Print/constants'
import { ButtonLink } from '~/components/common/Button'
import { getDomainPath } from '~/helpers/url'
import { PageHeader } from '~/components/Page'

type Props = {
  t: string => string,
  id: string
}

const Header = ({ t, id }: Props) => (
  <PageHeader
    actions={
      <React.Fragment>
        <ButtonLink
          to={`${getDomainPath()}/menu/${id}/print?template=${WEEKLY_MENU_TEMPLATE}`}
        >
          <FontAwesomeIcon icon={faFileExport} />
          {t('planning.header.actions.export.consumers')}
        </ButtonLink>
        <ButtonLink
          to={`${getDomainPath()}/menu/${id}/print?template=${INSTRUCTIONS_TEMPLATE}`}
        >
          <FontAwesomeIcon icon={faFileExport} />
          {t('planning.header.actions.export.chefs')}
        </ButtonLink>
      </React.Fragment>
    }
    title={t('planning.title')}
  />
)

export default withNamespaces()(Header)
