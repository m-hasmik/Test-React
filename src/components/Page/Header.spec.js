import React from 'react'
import { cleanup, render } from 'react-testing-library'

import { Header as PageHeader } from './Header'

const renderComponent = props => render(<PageHeader {...props} />)

describe('<PageHeader /> shallow', () => {
  afterEach(cleanup)

  it('contains two sections', () => {
    const { container } = renderComponent()
    const sectionsNumber = container.querySelectorAll(
      '[data-testid="header-section"]'
    ).length
    expect(sectionsNumber).toBe(2)
  })

  it('matches the snapshot', () => {
    const { container } = renderComponent({ title: 'Hello World' })
    expect(container).toMatchSnapshot()
  })
})
