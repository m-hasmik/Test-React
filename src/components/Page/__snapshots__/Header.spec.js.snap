// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> shallow matches the snapshot 1`] = `
.c2 {
  font-family: 'Poppins',sans-serif;
  font-weight: 400;
  font-size: 1.75rem;
  color: #001529;
}

.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background: #FFFFFF;
  border-bottom: 1px solid #E8E8E8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem 2rem 1rem;
  width: 100%;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="header-section"
    >
      <h1
        class="c2"
      >
        Hello World
      </h1>
    </div>
    <div
      class="c3"
      data-testid="header-section"
    />
  </div>
</div>
`;
